_posts/2022/2022-03-07-where-first-vs-find-by-know-what-your-scopes-are-doing.md

```diff
@@ -0,0 +1,78 @@
+---
+title: "Know what your scopes are doing: where.first vs find_by"
+description: "Attack of the abstracted SQL"
+layout: article
+category: ruby
+image:
+  base: "2022/where-first-vs-find-by-know-what-your-scopes-are-doing"
+  alt: "Wooden file card drawers"
+  credit: "Jan Antonin Kolar"
+  source: "https://unsplash.com/photos/lRoX0shwjUQ"
+
+---
+
+There are numerous benefits to using an Object-Relational Mapper (ORM) such as Active Record, to generate your SQL. You gain from clear repeatable abstractions, saving time and gaining readability.
+
+Unfortunately the very abstractions that make an ORM useful can also have unintended consequences.
+
+It is very easy to introduce performance problems as the _actual_ SQL that runs on your database server is generated by the framework.
+
+We recently came across an issue at CoverageBook that highlighted this very issue.
```

Comment:
Repetition of 'issue' doesn't work here. "We recently came across something..." Or. "...highlighted this very point." Or something like that.